src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -568,5 +580,17 @@ private void CheckSize(int countToAdd)
                 }
             }
         }
+
+        private struct EncodingPreamblePair
```

Comment:
`struct` is fine in this case, it's actually better than `class` as it avoids an additional indirection. `struct` would have been an issue if it was large and something like `List<>` would have been used instead of an array.